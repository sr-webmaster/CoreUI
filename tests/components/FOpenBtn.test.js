import { mount } from '@vue/test-utils'
import Component from '@freshinup/core-ui/src/components/FOpenBtn.vue'

describe('FOpenBtn', () => {
  describe('Snapshot', () => {
    it('should matched default', async () => {
      const wrapper = mount(Component, {
        propsData: {
          openText: 'Hide',
          closeText: 'Show'
        }
      })
      expect(wrapper.element).toMatchSnapshot()
    })

    it('should matched opening', async () => {
      const wrapper = mount(Component, {
        propsData: {
          openText: 'Hide',
          closeText: 'Show',
          isOpen: true
        }
      })
      expect(wrapper.element).toMatchSnapshot()
    })

    it('should matched styles', async () => {
      const wrapper = mount(Component, {
        propsData: {
          openText: 'Hide',
          closeText: 'Show',
          color: 'primary',
          flat: true
        }
      })
      expect(wrapper.element).toMatchSnapshot()
    })
  })

  describe('Method', () => {
    describe('toggle', () => {
      it('should toggle is opening', async () => {
        const wrapper = mount(Component, {
          propsData: {
            isOpen: true
          }
        })

        expect(wrapper.vm.isOpening).toEqual(true)

        wrapper.vm.toggle()

        expect(wrapper.vm.isOpening).toEqual(false)
      })

      it('should emit toggle with is opening', async () => {
        const wrapper = mount(Component, {
          propsData: {
            isOpen: true
          }
        })

        wrapper.vm.toggle()

        expect(wrapper.emitted('change')).toBeTruthy()
        expect(wrapper.emitted('change')[0][0]).toEqual(false)
      })
    })
  })
})
