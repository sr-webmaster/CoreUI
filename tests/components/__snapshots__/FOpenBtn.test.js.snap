// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FOpenBtn Snapshot should matched default 1`] = `
<div>
  <button
    class="v-btn theme--light -text--text"
    type="button"
  >
    <div
      class="v-btn__content"
    >
      <i
        aria-hidden="true"
        class="v-icon material-icons theme--dark"
      >
        keyboard_arrow_right
      </i>
       
      <span>
        Show
      </span>
    </div>
  </button>
   
  <div
    style="display: none;"
  />
</div>
`;

exports[`FOpenBtn Snapshot should matched opening 1`] = `
<div>
  <button
    class="v-btn theme--light -text--text"
    type="button"
  >
    <div
      class="v-btn__content"
    >
      <i
        aria-hidden="true"
        class="v-icon material-icons theme--dark"
      >
        keyboard_arrow_down
      </i>
       
      <span>
        Hide
      </span>
    </div>
  </button>
   
  <div />
</div>
`;

exports[`FOpenBtn Snapshot should matched styles 1`] = `
<div>
  <button
    class="v-btn v-btn--flat theme--light primary--text"
    type="button"
  >
    <div
      class="v-btn__content"
    >
      <i
        aria-hidden="true"
        class="v-icon material-icons theme--dark"
      >
        keyboard_arrow_right
      </i>
       
      <span>
        Show
      </span>
    </div>
  </button>
   
  <div
    style="display: none;"
  />
</div>
`;
